components/EventCard/EventCard.vue

```diff
@@ -1,164 +1,129 @@
 <template>
-  <div class="event-card">
-    <div class="event-content">
-      <div class="image">
-        <event-banner-image :src="bannerUrl" />
-      </div>
-      <div class="event-content-wrap">
-        <h3>
-          <nuxt-link
-            :to="{ name: 'events-eventId', params: { eventId: event.sys.id } }"
-          >
-            {{ event.fields.title }}
-          </nuxt-link>
-        </h3>
-        <div class="subtitle">
-          <div>{{ event.fields.location }}</div>
-          <!-- <div>{{ startDate }} - {{ endDate }}</div> -->
-        </div>
-        <div class="details">
-          <div class="detail">
-            {{ event.fields.summary }}
-          </div>
-          <div class="detail eventinfo">
-            {{ event.fields.eventType }} - SPARC Attendees:
-            {{ event.fields.sparcAttendees }}
-          </div>
-        </div>
-      </div>
+  <div class="upcoming-event">
+    <div class="upcoming-event__image">
+      <img :src="eventImage(event)" :alt="eventAlt(event)" />
+      <span>{{ event.fields.eventType }}</span>
+    </div>
+    <h3>
+      <a :href="event.fields.url" target="_blank">
+        {{ event.fields.title }}
+      </a>
+    </h3>
+    <div class="upcoming-event__detail">
+      <svg-icon name="icon-calendar" height="16" width="16" />
+      <p>{{ eventDate(event) }}</p>
+    </div>
+    <div class="upcoming-event__detail">
+      <svg-icon name="icon-map" height="16" width="16" />
+      <p>{{ event.fields.location }}</p>
     </div>
   </div>
 </template>
 
 <script>
-import { format, parseISO } from 'date-fns'
 import { pathOr } from 'ramda'
-import eventBannerImage from '@/components/EventBannerImage/EventBannerImage.vue'
+
 import FormatDate from '@/mixins/format-date'
 
 export default {
   name: 'EventCard',
-  components: {
-    eventBannerImage
-  },
+
   mixins: [FormatDate],
+
   props: {
     event: {
       type: Object,
-      default: () => {
-        return {
-          banner: '',
-          description: '',
-          name: '',
-          image: {
-            fields: {
-              file: {
-                url: ''
-              }
-            }
-          }
-        }
-      }
+      default: () => {}
     }
   },
-  computed: {
-    bannerUrl: function() {
-      return pathOr(
-        '',
-        ['fields', 'image', 'fields', 'file', 'url'],
-        this.event
-      )
+
+  methods: {
+    /**
+     * Get event image
+     * @returns {String}
+     */
+    eventImage: function(event) {
+      return pathOr('', ['fields', 'image', 'fields', 'file', 'url'], event)
```

Comment:
Without `pathOr`, it wasn't working. Would I do something like `(event.fields.image || '') || fields.image.fields.file.url` or something like it? This technique was used in another place in the app.

Reply:
I just ran into this issue when working with the resources page. I included a `v-show` in my img tag to only have it appear if there is an image, otherwise it won't appear. Would have to look at the rest of your code for context, but I assume this is the issue you're running into.

Reply:
Exactly the same problem. Which is weird because all of this data is being populated in `asyncData`, which is before the component event mounts. I confirmed that with the "Disable JavaScript" setting, the images are still being displayed.

Reply:
So I was looking at the data and I realized that some of mine don't even have images, which is why that issue was happening with me. Maybe it's the same case with events?

Reply:
Definitely the same for me. That's what `pathOr` is doing, but I guess it all depends on where it's breaking. @matthewtgilbride any insight on how to remove `pathOr` for this with `image` being not returned sometimes?

Reply:
In TS you'd handle this with the optional chaining operator (thing.property?.mightBeNullPropery.etc.etc)

I'd actually say this is a more valid use case for pathOr, since there is truly a property in the middle of a chain that might be undefined.

That being said...

You could use the ternary operator if only "image" is undefined.  I'm on my phone but will edit the comment later with an example.

Reply:
@cameronbaney @mahamzayed based on the code I'm assuming `event` is something like this crazy boi:
```
Entry<{ image?: Entry<{ file: { url: string } }> }>
```
Which unpacks to this:
```
{
  sys: Sys;
  fields: {
    image?: {
      sys: Sys;
      fields: {
        file: {
          url: string
        }
      }
    }
  }
}
```
So the underlying presumption here is that `image` is the only key that might be undefined.  If that's not the case this would be slightly different.

In TS:
`return event.fields.image?.fields.file.url ?? ''`

In JS with a ternary operator:
```
return event.fields.image 
  ? event.fields.image.file.url
  : ''
```

All that being said, I think it's debatable whether the `pathOr` approach or the ternary operator approach is better here.  The upside of using the ternary operator is that it makes it clear which property might be undefined.  But I don't think it's necessarily more readable than the original.

My beef with ramda is more with using `compose` to create multi step transformations that have to be read backwards...when the same could be done with vanilla JS.  For some ramda operators I've seen, there simply isn't a vanilla JS equivalent.  But for a lot of them there is an equivalent that I find much more readable.

@mahamzayed, Cam and I did some pairing on Typescript yesterday while you were in a meeting.  I hope it helped and I'd be happy to do the same with you next week whenever you have a few minutes if you'd like.

Reply:
@matthewtgilbride yes that would be really helpful! I'll be remote on Tuesday but we can sync up sometime on Wednesday. I've seen some of the typescript work you've done in several PRs now and would really like to get a better understanding on how it all ties together.